src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs

```diff
@@ -6,18 +6,32 @@ namespace System.Globalization
 {
     internal static partial class GlobalizationMode
     {
+        // Order of these properties matter because GetUseIcuMode is dependent on Invariant.
+        // So we need Invariant to be initialized first.
         internal static bool Invariant { get; } = GetGlobalizationInvariantMode();
 
-        // GetInvariantSwitchValue calls CLRConfig first to detect if the switch is defined in the config file.
+#if TARGET_WINDOWS
```

Comment:
We have .Unix.cs and .Windows.cs files for this. Why not move this to those files?

Reply:
Ah ok, this is because of the feedback I had about the order.

Reply:
So that the properties were in the same file for the initialization order. The other way would be to move Invariant and duplicate it to the `Unix/Windows` files. Is that what you'd like.

Reply:
> Ah ok, this is because of the feedback I had about the order.

Yeah, but I can instead duplicate Invariant to avoid the #ifdef -- would you prefer that?

Reply:
Yeah, I think avoiding the ifdef and duplicating a few lines between Windows/Unix would look better.

Reply:
Sounds good. I'll do that.